src/naive/date/mod.rs

```diff
@@ -99,7 +100,7 @@ mod tests;
 )]
 #[cfg_attr(feature = "rkyv-validation", archive(check_bytes))]
 pub struct NaiveDate {
-    yof: i32, // (year << 13) | of
+    yof: NonZeroI32, // (year << 13) | of
```

Comment:
Does this change the `NaiveDate` `MAX` or `MIN`? If not, why not?

Reply:
This doesn't change that much
Our internal representation uses an ordinal that is always in the range 1..366, and year flags that are never 0. So the value was already never 0; we now tell the type system as much.